### DISCUSSION TITLE
Issue credential

### DISCUSSION THREAD

**Discussion by RobertoSnap** [decentralized-identity/veramo#569] [Q&A] [answered]
Trying to issue a credential.

https://github.com/BROKLab/auth-bot/blob/veramo/src/network/veramo.service.ts#L95

On agent.keyManagerSignJWT() i get Error thrown _**"credentialSubject must not be empty"**_ but i dont see anything about crendentialSubject in the types. Am i doing this wrong?

**Comment by mirceanis** [comment, marked as answer]
I suspect that the error is coming from the [`createVerifiableCredential()` call](https://github.com/BROKLab/auth-bot/blob/ad00082b2e73ee790ed5835bd0d4389fa1a23dbd/src/network/veramo.service.ts#L104)
It is expecting that `credentialSubject` should have some `claims` (other properties beside `id`)

Using your example, perhaps the `data` parameter is intended to contain the claims, in which case it would look something like this:
```ts
credentialSubject: {
  id: subjectDidId,
  ...data
},
```

Take a look at the example in this other discussion: https://github.com/uport-project/veramo/discussions/571#discussioncomment-873268 too.

Please note that `createVerifiableCredential()` internally calls `keyManagerSign`. You don't have to call it on your own unless you need a JWS signature for other purposes.

**Reply by RobertoSnap** [reply]
Yea, worked when i added some data to credentialSubject 👍
